src/librustc_resolve/lib.rs

```diff
@@ -392,16 +392,20 @@ fn resolve_struct_error<'b, 'a: 'b, 'c>(resolver: &'b Resolver<'a>,
                              "`self` is not available in a static method. Maybe a `self` \
                              argument is missing?")
         }
-        ResolutionError::UnresolvedName(path, msg, context) => {
+        ResolutionError::UnresolvedName(path, msg, context, is_static_method, is_field) => {
             let mut err = struct_span_err!(resolver.session,
                                            span,
                                            E0425,
                                            "unresolved name `{}`{}",
                                            path,
                                            msg);
-
             match context {
-                UnresolvedNameContext::Other => { } // no help available
+                UnresolvedNameContext::Other => {
+                    if msg.is_empty() && is_static_method && is_field {
+                        err.help("This is a static method, so you don't have access to \
+                                  this type's fields");
```

Comment:
Personaly I'd remove the ‶so″ in the sentence.
Plus, I'd bit a bit more careful because there will be a lot of cases where the intent of the user wouldn't have been to access the struct fields at all. Maybe stay a bit more conditional : ‶If you meant to access self.{}, you can't because this is a static method.″ Well this is poorly-written but you've got the idea.